Text/TextQualityWatchdog/Util/TikaExtract.py

```diff
@@ -0,0 +1,159 @@
+# send requests to the Tika server, and save the extracted text files in a destination directory.
+# Currently, non-English files are filtered out.
+###
+
+import os
+import io
+import pickle
+
+from tika import parser
+from tika import language
+from azure.storage import blob
+from azure.storage.blob import BlobServiceClient, BlobClient, ContainerClient, __version__
+
+print("Azure Blob Storage v" + __version__ + " - Tika extraction handler\n")
+
+# Root path of this script
+APP_ROOT = os.path.dirname(os.path.abspath(__file__))
+APP_ROOT_STR = APP_ROOT.__str__()
+
+# Destination directory to save extracted text files to
+DEST_DIR = 'ExtractedTextData'
+
+# Download directory that we save unstructured files from blob storage to
+DOWNLOAD_DIR = 'CommonCrawlData'
+
+# Metadata directory that we save the tika-eval metrics to in a pickle file
+METADATA_DIR = 'Metadata'
+
+# Metrics directory where we save metrics used to summarize the run
+METRICS_DIR = 'Metrics'
+
+# Retrieve the connection string for use with the application. The storage
+# connection string is stored in an environment variable on the machine
+# running the application called AZURE_STORAGE_CONNECTION_STRING.
+connect_str = os.getenv('AZURE_STORAGE_CONNECTION_STRING')
+
+# Create the BlobServiceClient object which will be used to create a container client
+blob_service_client = BlobServiceClient.from_connection_string(connect_str)
+
+# Get the container client corresponding to the commoncrawl container
+cont_client = blob_service_client.get_container_client('commoncrawl3-refetched')
+
+# Get the list of all the blobs inside the commoncrawl container
+blob_list = cont_client.list_blobs()
+
+# Read the high water mark
+last_read_file_name = None
+last_read_file_path = APP_ROOT_STR + "\\" + METRICS_DIR + "\\lastread.txt"
+if os.path.exists(last_read_file_path):
+    with open(last_read_file_path, "r") as last_read_file:
+        last_read_file_name = last_read_file.readline()
+
+# Read the metrics dictionary, or create it if it does not exist
+metrics_dict = None
+metrics_dict_file_path = APP_ROOT_STR + "\\" + METRICS_DIR + "\\dataset_metrics.pkl"
+if os.path.exists(metrics_dict_file_path):
+    with open(metrics_dict_file_path, "rb") as metrics_dict_file:
+        metrics_dict = pickle.load(metrics_dict_file)
+else:
+    metrics_dict = {
+        "success" : 0, # English files that had text successfully extracted
+        "nonEng" : 0, # Files that were not in English
+        "failed" : 0, # Files that failed to have text extracted
+        "total" : 0 # Total count of the files
+    }
+
+# Process each blob
+for cur_blob in blob_list:
+    try:
+        # Get the name of the current blob
+        blobname = cur_blob.name
+        filename = blobname.split('/')[1]
+        print(blobname)
+
+        # Iterate until we hit the high water mark
+        if last_read_file_name != None and filename < last_read_file_name:
```

Comment:
That's a great point, Matt! I designed this tool to specifically work with our CommonCrawl dataset, which is static. Since this isn't meant to be a general-purpose tool, it's safe to keep track of the last read file for our purposes. I've added a comment in the file header to be more clear about the use case.